components/wishlist/WishlistButton/WishlistButton.tsx

```diff
@@ -32,7 +32,9 @@ const WishlistButton: FC<Props> = ({
     // @ts-ignore Wishlist is not always enabled
     (item) =>
       item.product_id === Number(productId) &&
-      (item.variant_id as any) === Number(variant.id)
+      (item.variant_id as any) === Number(variant.id) ||
+      item.product_id === productId &&
+      (item.variant_id as any) === variant.id
```

Comment:
Just doing the checks without Number casting should be ok

Reply:
Ok perfect! I will edit shortly

Reply:
Done